hackathon_site/dashboard/frontend/src/components/teamDetail/SimpleOrderTables/SimpleOrderTables.tsx

```diff
@@ -0,0 +1,233 @@
+import { mockPendingOrdersInTable, mockReturnedOrdersInTable } from "testing/mockData";
+import React, { useState } from "react";
+import Container from "@material-ui/core/Container";
+import styles from "components/general/OrderTables/OrderTables.module.scss";
+import { Checkbox, FormGroup, Grid, TableCell, Tooltip } from "@material-ui/core";
+import Button from "@material-ui/core/Button";
+import {
+    GeneralOrderTitle,
+    GeneralPendingTable,
+    GeneralReturnTable,
+} from "components/general/OrderTables/OrderTables";
+import {
+    Field,
+    FieldProps,
+    Form,
+    Formik,
+    FormikValues,
+    useFormikContext,
+} from "formik";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import { OrderInTable } from "api/types";
+
+interface SimpleOrderFormValues {
+    itemIdsChecked: string[];
+    checkAll: boolean;
+}
+
+const TableCheckbox = ({
+    field,
+    option,
+    order,
+    ...props
+}: FieldProps & { option: string; order: OrderInTable }) => {
+    const { values: formikValues, setFieldValue } =
+        useFormikContext<SimpleOrderFormValues>();
+
+    const recheckAllItems = (checked: boolean) => {
+        if (order.hardwareInTableRow.length - formikValues.itemIdsChecked.length <= 1) {
+            setFieldValue("checkAll", checked);
+        }
+    };
+
+    return (
+        <FormGroup {...field} {...props}>
+            <FormControlLabel
+                name={field.name}
+                value={option}
+                control={<Checkbox color="primary" />}
+                label=""
+                checked={field.value?.includes(option)}
+                onChange={(e, checked) => {
+                    field.onChange(e);
+                    recheckAllItems(checked);
+                }}
+            />
+        </FormGroup>
+    );
+};
+
+const TableCheckAll = ({
+    field,
+    order,
+    ...props
+}: FieldProps & { order: OrderInTable }) => {
+    const { setFieldValue } = useFormikContext<SimpleOrderFormValues>();
+
+    const checkAllItems = (checked: boolean) =>
+        setFieldValue(
+            "itemIdsChecked",
+            checked ? order.hardwareInTableRow.map((item) => `hardware-${item.id}`) : []
+        );
+
+    return (
+        <FormGroup {...field} {...props}>
+            <FormControlLabel
+                name={field.name}
+                value={field.name}
+                onChange={(e, checked) => {
+                    field.onChange(e);
+                    checkAllItems(checked);
+                }}
+                control={<Checkbox color="primary" />}
+                label=""
+                checked={field.value}
+            />
+        </FormGroup>
+    );
+};
+
+const getExtraColumn = (order: OrderInTable) => ({
```

Comment:
I don't think this name is descriptive enough because it makes it sound like the checkbox column is just an extra irrelevant column. Maybe call it something like enableCheckboxColumn to understand what column is being added as well as putting importance to it

Reply:
I orginally named it this way bc I wanted SimpleOrderTables to be generalized so this extra column could be anything, but now I don't know if that would happen 🤔 I'll change it to a more descriptive name and add a comment but I'd still like to keep it as a general extra column.